streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -106,7 +107,7 @@ public StreamsProducer(final StreamsConfig config,
                 producerConfigs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, applicationId + "-" + taskId);
 
                 eosBetaProducerConfigs = null;
-
+                verifyTransactionTimeoutCompatibility(producerConfigs, config);
```

Comment:
It might be better to do this check in `StreamsConfig` ?

Reply:
+1, we should fail-fast at the level of the KafkaStreams. Otherwise in eos-alpha we actually have to wait for all the individual StreamThreads to be created, started, complete a rebalance to get a task assignment, and _then_ start throwing IllegalArgumentException on each thread when it goes to create a task producer. (Can we also log an error in addition to the IllegalArgumentException?)

Reply:
Sounds good!